testdata/project-v3-config-gen/go.mod

```diff
@@ -0,0 +1,74 @@
+module sigs.k8s.io/kubebuilder/testdata/project-v3-config-gen
+
+go 1.17
+
+require (
```

Comment:
In the go.mod of Kubebuilder, we see that we have deps that are used by config-gen only.
How/When these deps are used?
Should we need to change the go.mod of the scaffold?
Are we doing things with the Kubebuilder CLI to make config-gen work?
How that would end up on the plugin subcommand? 

c/c @pwittrock